plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2051,6 +2066,8 @@ So if getMinSpeed() returns null we fall back to getSpeed().
         devices.addDevice(new ChannelDef(_qemuGuestAgentSocketName, ChannelDef.ChannelType.UNIX,
                           new File(_qemuSocketsPath + "/" + vmTO.getName() + "." + _qemuGuestAgentSocketName)));
 
+        devices.addDevice(new WatchDogDef(_watchDogAction, _watchDogModel));
```

Comment:
@wido the default action `NONE` is acceptable as adding a watchdog device that might possible reset the instance is not acceptable for everyone. Also, can there be a case where the default watchdog model `I6300ESB` is not available, in which case can we initialize this as null and add the device only when _watchDogModel is not null and read options from agent.properties file /cc @DaanHoogland @nvazquez @ustcweizhou @remibergsma what do you think?

Reply:
As far as I know that's not pissible @rhtyd 

These models are always available in Qemu